apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -0,0 +1,360 @@
+            .onMalformedInput(CodingErrorAction.REPORT)
+            .onUnmappableCharacter(CodingErrorAction.REPORT);
+
+    CharsetEncoder outEnc =
+        StandardCharsets.UTF_8
+            .newEncoder()
+            .onMalformedInput(CodingErrorAction.REPORT)
+            .onUnmappableCharacter(CodingErrorAction.REPORT);
+
+    try (FileInputStream is = new FileInputStream(originalNpiDataFile.toFile().getAbsolutePath());
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is, inDec));
+        FileOutputStream fw =
+            new FileOutputStream(convertedNpiDataFile.toFile().getAbsolutePath());
+        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(fw, outEnc))) {
+
+      // Get indexes according to header
+      String line = reader.readLine();
+      String[] fields = line.split(",");
+      Map<String, Integer> indexes = getIndexNumbers(fields);
+      Integer npiIndex = getIndexNumberForField(indexes, "NPI");
+      Integer entityTypeCodeIndex = getIndexNumberForField(indexes, "Entity Type Code");
+      Integer orgNameIndex =
+          getIndexNumberForField(indexes, "Provider Organization Name (Legal Business Name)");
+
+      while ((line = reader.readLine()) != null) {
+        fields = line.split(",");
+
+        String orgName = fields[orgNameIndex].trim().replace("\"", "");
+        String npi = fields[npiIndex].trim().replace("\"", "");
+        String entityTypeCode = fields[entityTypeCodeIndex].trim().replace("\"", "");
+
+        // entity type code 2 is organization
+        if (!Strings.isNullOrEmpty(entityTypeCode) && Integer.parseInt(entityTypeCode) == 2) {
+          out.write(npi + "\t" + orgName);
+          out.newLine();
+        }
+      }
+    }
+  }
+
+  /**
+   * sets the index number to their header value.
+   *
+   * @param fields the string array of header values.
+   * @return map of indexes and field names
+   */
+  private static Map<String, Integer> getIndexNumbers(String[] fields) {
+    Map<String, Integer> indexNumbers = new HashMap<String, Integer>();
+    Integer indexCounter = 0;
+    for (String field : fields) {
+      indexNumbers.put(field.trim().replace("\"", ""), indexCounter++);
+    }
+
+    return indexNumbers;
+  }
+
+  /**
+   * gets the index number according to their field name.
+   *
+   * @param fieldName the header value to check.
+   * @param indexNumbers the map of header values and indexes.
+   * @return int of index
+   */
+  private static int getIndexNumberForField(Map<String, Integer> indexNumbers, String fieldName) {
+    if (indexNumbers.containsKey(fieldName)) {
+      return indexNumbers.get(fieldName);
+    }
+
+    throw new IllegalStateException(
+        "NPI Org File Processing Error: Cannot field fieldname " + fieldName);
+  }
+
+  /**
+   * Extracts a file name.
+   *
+   * @param getMonthBefore gets the month before
+   * @return a file name string
+   */
+  private static String getFileName(boolean getMonthBefore) {
```

Comment:
on the last version of the PR I thought we agreed to not have this build know anything about the file and whether it exists or whether we have a new one, etc. That was going to go in the jenkins job. Was there a discussion to the contrary?